### DISCUSSION TITLE
Change marker colour by country V3

### DISCUSSION THREAD

**Discussion by rikgale** [vradarserver/vrs#96] [General]
Hello 
I am using V3 preview 11. Is it possible to change the marker colour by which country the aircraft is from?
Currently I can change the marker colour based on operator, MLaT status, Military and ICAO Type code, but do not seem to be able to do it by Country or ModeSCountry.
Any pointers welcomed.
Thanks,
rikgale

**Comment by rikgale** [comment]
After lots of playing about it turns out you can interact with the marker based on ``aircraft.country`` and not ``aircraft.modescountry`` as I had been trying. I can now set marker colour based on country recorded inDB

How do you interact with the isInteresting valve of an aircraft?

```
if(aircraft.isInterested == '1'){	
return "#FF0000"; // red
}
```

does not seem to work

```
if(aircraft.isInterested.val == '1'){	
return "#FF0000"; // red
}
```

Throws an issue at the use of val

```
if(aircraft.isInterested){	
return "#FF0000"; // red
}
```
no joy

replacing ``isInterested`` with ``isInteresting`` yeilds the same results as above.

Any suggestions grateful welcomed.

**Reply by vradarserver** [reply]
Sorry for the delay in replying. If you take a look the aircraft class here then the fields are declared near the start of the class:

https://github.com/vradarserver/vrs/blob/master/VirtualRadar.WebSite/Site/Web/script/vrs/aircraft.ts

(as of Mar 2023 they start at line 364 here: https://github.com/vradarserver/vrs/blob/master/VirtualRadar.WebSite/Site/Web/script/vrs/aircraft.ts#L364)

Neither isInterested or isInteresting exist, you want `userInterested` instead, as in:

```
if(aircraft.userInterested.val) {
    // do something when the flag is set
}
```

The names of the fields in the JSON are kept kind-of short to try to reduce the bandwidth that the JSON can consume. t's normal for the aircraft property names to differ significantly from the names of the JSON fields that they're filled from.

All of the non-array fields are an object with two properties, `chg` (true if the value changed in the last update from the server, otherwise false) and `val` (the value last sent by the server). The `val` property can be null if the value has never been sent for the aircraft.

**Comment by rikgale** [comment]
Thank you so much. That is exactely the information I have been looking for.

**Comment by rikgale** [comment]
Follow on question:
I've got a script that changes the marker shape and colour based on various parameters for V3... 
https://github.com/rikgale/VRSCustomMarkers/blob/main/MyMarkers1.html
...
How would I go about changing it so that out the marker outline colour could be changed for SVGs. This has frustrated me so far and any guidance would be appreciated. Thank you.

**Reply by vradarserver** [reply]
Sorry for the delay in replying - there's no support in the code for changing the stroke colour for the fill paths. Feel free to add it as an issue but at the moment any VRS time I get is being spent on the .NET Core conversion, so it might be a while before it gets done.

**Reply by rikgale** [reply]
Thank you. I will add as an issue and wait patiently :)
